main_training.py

```diff
@@ -54,10 +54,15 @@ def main(**kwargs):
     llama_config = get_model_config(cfg.model_variant)
 
     if cfg.low_cpu_fsdp:
-        with torch.device("meta"):
+        if rank == 0:
             model = LLaMA(llama_config)
+            model.reset_parameters()
```

Comment:
I'd kinda prefer to see this gated `if hasattr(model, 'reset_parameters')`, to make the code more extensible to other model types, though I guess we are just explicitly getting LLaMA and not using get_model.

Reply:
agreed.  I had been deciding between the two for quite a while, the get_model() currently seems to be a little too general for this purpose and potentially hide from user on "oh, the code is not using the constructor itself, but a specific get_model, maybe something more is involved in their implementation on initializting the model?"  

So at early stage (when we only target Llama), I want to, for now, make this more clear that we are simply create the vanilla model using the constructor.

Once we extend this repo to true fms model"s", then I prefer to make the swtich to use get_model